File: backend/app/schemas/clothing.py
from enum import Enum

from pydantic import BaseModel

class ClothingCategory(str, Enum):
    INNER_TOP = "INNER_TOP"
    OUTER_TOP = "OUTER_TOP"
    BOTTOM = "BOTTOM"
    SHOES = "SHOES"


class ClothingFormality(str, Enum):
    CASUAL = "CASUAL"
    BUSINESS_CASUAL = "BUSINESS_CASUAL"
    FORMAL = "FORMAL"
    ATHLETIC = "ATHLETIC"


class ClothingColor(str, Enum):
    RED = "RED"
    ORANGE = "ORANGE"
    YELLOW = "YELLOW"
    GREEN = "GREEN"
    BLUE = "BLUE"
    PURPLE = "PURPLE"
    BLACK = "BLACK"
    WHITE = "WHITE"
    BROWN = "BROWN"
    GREY = "GREY"
    PINK = "PINK"
    MULTI = "MULTI"

class ClothingItem(BaseModel):
    name: str
    description: str | None = None
    category: ClothingCategory
    formality: ClothingFormality
    color: ClothingColor

File: web/src/types/clothing.ts
export type ClothingCategory =
  | "INNER_TOP"
  | "OUTER_TOP"
  | "BOTTOM"
  | "SHOES"

export type ClothingColor =
  | "RED"
  | "ORANGE"
  | "YELLOW"
  | "GREEN"
  | "BLUE"
  | "PURPLE"
  | "BLACK"
  | "WHITE"
  | "BROWN"
  | "GREY"
  | "PINK"
  | "MULTI"

export type ClothingFormality =
  | "CASUAL"
  | "BUSINESS_CASUAL"
  | "FORMAL"
  | "ATHLETIC"

export interface ClothingItem {
  id: number
  name: string
  description: string | null
  category: ClothingCategory
  formality: ClothingFormality
  color: ClothingColor
  image_url: string
}

export interface ClothingFilters {
  categories: ClothingCategory[]
  colors: ClothingColor[]
  formalities: ClothingFormality[]
}

export const CLOTHING_CATEGORIES: ClothingCategory[] = [
  "INNER_TOP",
  "OUTER_TOP",
  "BOTTOM",
  "SHOES",
]

export const CLOTHING_COLORS: ClothingColor[] = [
  "RED",
  "ORANGE",
  "YELLOW",
  "GREEN",
  "BLUE",
  "PURPLE",
  "BLACK",
  "WHITE",
  "BROWN",
  "GREY",
  "PINK",
  "MULTI",
]

export const CLOTHING_FORMALITIES: ClothingFormality[] = [
  "CASUAL",
  "BUSINESS_CASUAL",
  "FORMAL",
  "ATHLETIC",
]